src/Contexts/AuthDialogContext/index.jsx

```diff
@@ -24,50 +27,81 @@ export const AuthProvider = (props) => {
         email: user.email,
         password: user.password,
       });
-      console.log("login requrest resp",res);
-      if(res.statusText==='OK'){
-        utils.setLocalStorage('authToken',res.data.encodedToken);
+      console.log("login requrest resp", res);
+      if (res.statusText === "OK") {
+        utils.setLocalStorage("authToken", res.data.encodedToken);
         setIsLoading(false);
-        setError({errorExists:false,errorMessage:''})
-        setUser(res.data.foundUser)
+        setError({ errorExists: false, errorMessage: "" });
+        setUser({
+          ...res.data.foundUser,
+          isAuthenticated: true,
+          encodedToken: res.data.encodedToken,
+        });
         return true;
       }
     } catch (e) {
-      setError({errorExists:true,errorMessage:"Invalid Credentials. Please try again"});
+      setError({
+        errorExists: true,
+        errorMessage: "Invalid Credentials. Please try again",
+      });
       setIsLoading(false);
     }
   };
 
   const logoutHandler = () => {
-    utils.removeLocalStorage('authToken')
-		setUser(null);
+    utils.removeLocalStorage("authToken");
+    setUser(null);
   };
-  
+
   const signUpHandler = async (user) => {
     setIsLoading(true);
-    try{
-      const res = await axios.post('/api/auth/signup',{
+    try {
+      const res = await axios.post("/api/auth/signup", {
         email: user.email,
         password: user.password,
-      })
-      if(res.statusText==='Created'){
-        setUser(res.data.createdUser)
-        utils.setLocalStorage('authToken',res.data.encodedToken);
-        setError({errorExists:false,errorMessage:''})
-        setIsLoading(false)
+      });
+      if (res.statusText === "Created") {
+        // setUser(res.data.createdUser)
+        setUser({
+          ...res.data.foundUser,
+          isAuthenticated: true,
+          encodedToken: res.data.encodedToken,
+        });
+        utils.setLocalStorage("authToken", res.data.encodedToken);
+        setError({ errorExists: false, errorMessage: "" });
+        setIsLoading(false);
         return true;
       }
-    }
-    catch(e){
-      setError({errorExists:true,errorMessage:"Something went wront. Please try again"});
+    } catch (e) {
+      setError({
+        errorExists: true,
```

Comment:
if there is no error simply keep the errorMessage to empty string.
There is no need of using errorExists

Reply:
thanks for the suggestion , I will keep this in mind moving forward